src/peers.py

```diff
@@ -130,105 +130,193 @@ def peerline(cls, line):
         if the line is a correctly-formatted peer line.
         """
         if cls.isnoiseline(line):
-            return False
+            return None
 
-        peertype = cls.tallytotype(line[0])
-        if peertype == 'unknown':
-            warn('Unknown peer tally code: %s' % line[0])
-            return False
+        # 10 comma-separated fields are chronyc
+        fields = line.split(',')
+        if len(fields) == 10:
+            fields = cls.chrony_peerline(fields)
+            if fields:
+                return cls.validate_peerfields(fields, False)
+            else:
+                return None
 
+        # 10 space-separated fields are ntpq
         fields = line[1:].split()
+        if len(fields) == 10:
+            fields = cls.ntpd_peerline(line[0], fields)
```

Comment:
You might prefer to use collections.namedtuple here, rather than constructing a dictionary. Or not; the code reads nicer, but they lack flexibility.

Reply:
Seems neat; might try it in future.